pulsar-broker-common/src/main/java/org/apache/pulsar/broker/web/AuthenticationFilter.java

```diff
@@ -71,10 +74,25 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 
             if (!isSaslRequest(httpRequest)) {
                 // not sasl type, return role directly.
-                String role = authenticationService.authenticateHttpRequest((HttpServletRequest) request);
+                String authMethodName = httpRequest.getHeader(PULSAR_AUTH_METHOD_NAME);
+                AuthenticationState authenticationState = null;
+                if (authMethodName != null && authenticationService.getAuthenticationProvider(authMethodName) != null) {
+                    authenticationState = authenticationService
+                            .getAuthenticationProvider(authMethodName).newHttpAuthState(httpRequest);
+                    request.setAttribute(AuthenticatedDataAttributeName, authenticationState.getAuthDataSource());
+                } else {
+                    request.setAttribute(AuthenticatedDataAttributeName,
+                            new AuthenticationDataHttps((HttpServletRequest) request));
+                }
+                String role;
+                if (authenticationState != null) {
```

Comment:
Nit: I think we could consolidate this conditional and the one above starting on line 79. That could help make it clear that the `else` block is a legacy use case to handle calls that are missing the `PULSAR_AUTH_METHOD_NAME` header. Do you agree?